contxt/cli/commands/services.py

```diff
@@ -1,23 +1,219 @@
-from typing import List
+from typing import List, Optional
 
 import click
 
 from contxt.cli.clients import Clients
 from contxt.cli.utils import fields_option, print_table, sort_option
-from contxt.models.contxt import Service
+from contxt.models.contxt import Service, ServiceGrant, ServiceScope
+from contxt.utils.serializer import Serializer
 
 
 @click.group()
 def services() -> None:
-    """Services."""
+    """Contxt Services"""
 
 
-@services.command()
+@services.group()
+def scopes() -> None:
+    """Service Scopes"""
+
+
+@services.group()
+def dependencies() -> None:
+    """Service Dependencies"""
+
+
+def _get_dependency_info_for_scopes(clients: Clients, from_service_id: int, to_service_id: int):
+
+    # retrieve the origin service
+    from_service = clients.contxt.get_service(from_service_id)
+    to_service = clients.contxt.get_service(to_service_id)
+
+    # retrieve the dependencies of that service
+    deps = clients.contxt.get_service_dependencies(from_service.id)
+
+    print(deps)
+    # find the dependency for this "to service"
+    to_dependency = next((g for g in deps if int(g.to_service_id) == int(to_service.id)), None)
+    if to_dependency is None:
+        print(
+            f"Service ID {from_service.id} ({from_service.name}) does not currently have a "
+            f"dependency on Service ID {to_service.id} ({to_service.name}). You can add it via "
+            f'the "contxt services dependencies add" CLI command'
+        )
+        return None, None, None
+
+    to_service_scopes = {s.label: s for s in clients.contxt.get_service_scopes(to_service.id)}
+    existing_dep_scopes = [e.label for e in to_dependency.ServiceScopes]
+
+    return to_service_scopes, existing_dep_scopes, to_dependency
+
+
+"""
+Services Commands
+"""
+
+
+@services.command("get")
 @click.argument("id", default="")  # HACK: make an optional argument
 @fields_option(default=["id", "name", "service_type", "description"], obj=Service)
 @sort_option(default="id")
 @click.pass_obj
-def get(clients: Clients, id: str, fields: List[str], sort: str) -> None:
+def get_services(clients: Clients, id: Optional[str], fields: List[str], sort: str) -> None:
     """Get service(s)"""
     items = [clients.contxt.get_service(id)] if id else clients.contxt.get_services()
-    print_table(items=items, keys=fields, sort_by=sort)
+    if id:
+        print(Serializer.to_pretty_cli(items[0]))
+    else:
+        print_table(items=items, keys=fields, sort_by=sort)
+
+
+"""
+Scopes Commands
+"""
+
+
+@scopes.command("get")
+@click.argument("service_id")
+@fields_option(default=["label", "description"], obj=ServiceScope)
+@sort_option(default="label")
+@click.pass_obj
+def get_scopes(clients: Clients, service_id: str, fields: List[str], sort: str) -> None:
```

Comment:
Can you add docstrings to these commands so we can get the description shown on `contxt services scopes get --help`, etc?